### DISCUSSION TITLE
How do I launch and wait for other projects

### DISCUSSION THREAD

**Discussion by jwilsonbrandt** [nrwl/nx#19598] [General]
I am trying to set up an nx project with nestjs using graphql (gateway) & subgraph apps. When debugging locally, I need my subgraph apps to start up before the main gateway app. 

I tried to use `dependsOn` in the gateway apps project.json config, and this will launch all the subgraph apps when I serve it, but it never launches the main gateway app (only the dependents). Is there another way to do this, or is there something I'm missing? 

`
"dependsOn": [
  "subgraph-app-1:serve",
  "subgraph-app-2:serve"
]
`

**Comment by Ambroos** [comment]
What I see used for this most often, adjusted for your needs, is `nx run-many --target serve --parallel=3 --projects gateway,subgraph-app-1,subgraph-app-2`. This gets a bit more complex if you have to build in some wait time but is not impossible by using a run command executor that just runs a wait command for X seconds first.

There is no built-in way for things in serve mode to report they're ready to accept requests.

**Reply by jwilsonbrandt** [reply]
Thank you for the kind reply! Much appreciated :)
